### ISSUE TITLE
yt-dlp in CentOS doesn't write video URL into WEBM while yt-dlp in Windows 7 does

### ISSUE THREAD

**Issue by pboettcher** [yt-dlp/yt-dlp#3513]
### Checklist

- [X] I'm reporting a bug unrelated to a specific site
- [X] I've verified that I'm running yt-dlp version **2022.04.08** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Description

I am calling yt-dlp on schedule from a bash script:

/usr/local/bin/yt-dlp -o "%(upload_date)s %(title)s.%(ext)s" --write-annotations --download-archive "$ArchiveFile" --add-metadata --write-sub --sub-lang $SubLang --write-auto-sub --max-downloads $MaxDownloads -f "bestaudio[ext=webm]" -i "$ChannelPath"

After checking the WEBM file dump I see that it doesn't have an URL inside. The similar command line executed for yt-dlp in Windows produces a WEBM file with URL inside. I noticed this issue already with latest youtube-dl and am not sure whether it is a bug or a problem with environment on my CentOS 7 server.

### Verbose log

```shell
[debug] Command-line config: ['-vU', '-o', '%(upload_date)s %(title)s.%(ext)s', '--write-annotations', '--download-archive', '/home/downloader/ytdl/Auswandern 2022-25.archive', '--add-metadata', '--write-sub', '--sub-lang', 'de', '--write-auto-sub', '--max-downloads', '1', '-f', 'bestaudio[ext=webm]', '-i', 'https://www.youtube.com/channel/UCZ4zZkE8REK8Ja805VNfl6w/videos']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8 (No ANSI), err UTF-8, pref UTF-8
[debug] yt-dlp version 2022.04.08 [7884ade65] (zip)
[debug] Python version 3.6.8 (CPython 64bit) - Linux-3.10.0-1160.49.1.el7.x86_64-x86_64-with-centos-7.9.2009-Core
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg 2.8.15 (needs_adtstoasc,fdk), ffprobe 2.8.15
[debug] Optional libraries: certifi, Crypto, sqlite
[debug] Proxy map: {}
[debug] Loading archive file '/home/downloader/ytdl/Auswandern 2022-25.archive'
[debug] [youtube:tab] Extracting URL: https://www.youtube.com/channel/UCZ4zZkE8REK8Ja805VNfl6w/videos
[debug] [youtube:tab] Final URL: https://www.youtube.com/channel/UCZ4zZkE8REK8Ja805VNfl6w/videos
[debug] [youtube] Extracting URL: https://www.youtube.com/watch?v=W4Vvua7H7zs
[debug] Sort order given by extractor: quality, res, fps, hdr:12, source, codec:vp9.2, lang, proto
[debug] Formats sorted by: hasvid, ie_pref, quality, res, fps, hdr:12(7), source, vcodec:vp9.2(10), acodec, lang, proto, filesize, fs_approx, tbr, vbr, abr, asr, vext, aext, hasaud, id
[debug] Downloading subtitles: de
[debug] Invoking downloader on "https://www.youtube.com/api/timedtext?v=W4Vvua7H7zs&caps=asr&xoaf=5&hl=en&ip=0.0.0.0&ipbits=0&expire=1650606392&sparams=ip%2Cipbits%2Cexpire%2Cv%2Ccaps%2Cxoaf&signature=395DF0CB3E51DBDB141A3FFBF63DA1CACF3EB604.E6F15155B0D1CC3012BABF261E9EF7A9EB937111&key=yt8&kind=asr&lang=de&fmt=vtt"
WARNING: There are no annotations to write.
[debug] Invoking downloader on "https://rr3---sn-8vq54voxqx-cxgs.googlevideo.com/videoplayback?expire=1650602793&ei=yd5hYt-qEL2qx_APo5em0A8&ip=XX.XX.XXX.XX&id=o-ABW1JgpjjtWFAwah5AYX4KN1nIOL4VjUibo9K04hj2rp&itag=251&source=youtube&requiressl=yes&mh=tz&mm=31%2C29&mn=sn-8vq54voxqx-cxgs%2Csn-i5heen7z&ms=au%2Crdu&mv=m&mvi=3&pcm2cms=yes&pl=24&initcwndbps=1313750&vprv=1&mime=audio%2Fwebm&gir=yes&clen=5714073&otfp=1&dur=425.541&lmt=1633902382523737&mt=1650580851&fvip=2&keepalive=yes&fexp=24001373%2C24007246&c=ANDROID&txp=6211224&sparams=expire%2Cei%2Cip%2Cid%2Citag%2Csource%2Crequiressl%2Cvprv%2Cmime%2Cgir%2Cclen%2Cotfp%2Cdur%2Clmt&sig=AOq0QJ8wRQIhAK2InDBPorkAYb_p3ge-wfF8OCy1msHElG5GDT7YW8HpAiBYLal08g9DcIBhJAMoy0dbmslE_5onj-N15o-EEF2wsQ%3D%3D&lsparams=mh%2Cmm%2Cmn%2Cms%2Cmv%2Cmvi%2Cpcm2cms%2Cpl%2Cinitcwndbps&lsig=AG3C_xAwRQIgCPSxcGaRuLoPNCzTiDsEdOB39w0-ePNS6hyfuS_8VVkCIQDlPdtDWDU2FOmvk0cj2ataSo5tk24eTCKcro4c7BFwxA%3D%3D"
[debug] ffmpeg command line: ffmpeg -y -loglevel repeat+info -i 'file:20211012 Auswandern - 49.  Landwirte, Landwirtschaft, Investitionen.webm' -map 0 -dn -ignore_unknown -c copy -write_id3v1 1 -metadata 'title=Auswandern: 49.  Landwirte, Landwirtschaft, Investitionen' -metadata date=20211012 -metadata 'description=Auswandern 2022 -2025 bei Youtube: https://www.youtube.com/channel/UCZ4zZkE8REK8Ja805VNfl6w - Paraguay, Brasilien, Uruguay, Argentinien, Georgien ...

Wie und wo lohnen sich Investitionen in der Landwirtschaft?' -metadata 'synopsis=Auswandern 2022 -2025 bei Youtube: https://www.youtube.com/channel/UCZ4zZkE8REK8Ja805VNfl6w - Paraguay, Brasilien, Uruguay, Argentinien, Georgien ...

Wie und wo lohnen sich Investitionen in der Landwirtschaft?' -metadata 'purl=https://www.youtube.com/watch?v=W4Vvua7H7zs' -metadata 'comment=https://www.youtube.com/watch?v=W4Vvua7H7zs' -metadata 'artist=Auswandern 2022 - 25' -movflags +faststart 'file:20211012 Auswandern - 49.  Landwirte, Landwirtschaft, Investitionen.temp.webm'
[debug] Adding to archive: youtube W4Vvua7H7zs
```

**Comment by pukkandan** [comment]
> After checking the WEBM file dump I see that it doesn't have an URL inside. 

How did you check this?

**Comment by pukkandan** [comment]
Also, that is not the complete log. Looks like you captured only stderr

**Comment by pboettcher** [comment]
> > After checking the WEBM file dump I see that it doesn't have an URL inside.
> 
> How did you check this?

Looked at hex dump in various encodings: ANSI, UTF-8, UTF-16 LE, UTF-16 BE. Unless the URL is encrypted, it is not there.
When I download the file in Windows, the URL is there. Also FFMPEG is then able to convert and pack it into MP3.

**Comment by pboettcher** [comment]
> Also, that is not the complete log. Looks like you captured only stderr

I used the -vU switch. Isn't that enough?

**Comment by pukkandan** [comment]
Can you run ffprobe on the file and show it's output?

```
ffprobe "20211012 Auswandern - 49.  Landwirte, Landwirtschaft, Investitionen.webm"
```

**Comment by pboettcher** [comment]
`[downloader@localhost Auswandern 2022-25]$ ffprobe ./20211012\ Auswandern\ -\ 49.\ \ Landwirte\,\ Landwirtschaft\,\ Investitionen.webm
ffprobe version 2.8.15 Copyright (c) 2007-2018 the FFmpeg developers
  built with gcc 4.8.5 (GCC) 20150623 (Red Hat 4.8.5-36)
  configuration: --prefix=/usr --bindir=/usr/bin --datadir=/usr/share/ffmpeg --incdir=/usr/include/ffmpeg --libdir=/usr/lib64 --mandir=/usr/share/man --arch=x86_64 --optflags='-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic' --extra-ldflags='-Wl,-z,relro ' --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libvo-amrwbenc --enable-version3 --enable-bzlib --disable-crystalhd --enable-gnutls --enable-ladspa --enable-libass --enable-libcdio --enable-libdc1394 --enable-libfdk-aac --enable-nonfree --disable-indev=jack --enable-libfreetype --enable-libgsm --enable-libmp3lame --enable-openal --enable-libopenjpeg --enable-libopus --enable-libpulse --enable-libschroedinger --enable-libsoxr --enable-libspeex --enable-libtheora --enable-libvorbis --enable-libv4l2 --enable-libx264 --enable-libx265 --enable-libxvid --enable-x11grab --enable-avfilter --enable-avresample --enable-postproc --enable-pthreads --disable-static --enable-shared --enable-gpl --disable-debug --disable-stripping --shlibdir=/usr/lib64 --enable-runtime-cpudetect
  libavutil      54. 31.100 / 54. 31.100
  libavcodec     56. 60.100 / 56. 60.100
  libavformat    56. 40.101 / 56. 40.101
  libavdevice    56.  4.100 / 56.  4.100
  libavfilter     5. 40.101 /  5. 40.101
  libavresample   2.  1.  0 /  2.  1.  0
  libswscale      3.  1.101 /  3.  1.101
  libswresample   1.  2.101 /  1.  2.101
  libpostproc    53.  3.100 / 53.  3.100
Input #0, matroska,webm, from './20211012 Auswandern - 49.  Landwirte, Landwirtschaft, Investitionen.webm':
  Metadata:
    title           : Auswandern: 49.  Landwirte, Landwirtschaft, Investitionen
    encoder         : Lavf56.40.101
  Duration: 00:07:05.54, start: 0.007000, bitrate: 107 kb/s
    Stream #0:0(eng): Audio: opus, 48000 Hz, stereo, fltp (default)
`

**Comment by pukkandan** [comment]
If that is all the output yopu are getting from ffprobe on windows, you are doing something wrong

```
❯ yt-dlp --add-metadata W4Vvua7H7zs
[youtube] W4Vvua7H7zs: Downloading webpage
[youtube] W4Vvua7H7zs: Downloading android player API JSON
[youtube] W4Vvua7H7zs: Downloading MPD manifest
[youtube] W4Vvua7H7zs: Downloading MPD manifest
[info] W4Vvua7H7zs: Downloading 1 format(s): 248+251
[download] Auswandern - 49.  Landwirte, Landwirtschaft, Investitionen [W4Vvua7H7zs].webm has already been downloaded
[Metadata] Adding metadata to "Auswandern - 49.  Landwirte, Landwirtschaft, Investitionen [W4Vvua7H7zs].webm"

❯ ffprobe "Auswandern - 49.  Landwirte, Landwirtschaft, Investitionen [W4Vvua7H7zs].webm"
ffprobe version N-106624-g391ce570c8-20220415 Copyright (c) 2007-2022 the FFmpeg developers
  built with gcc 11.2.0 (crosstool-NG 1.24.0.533_681aaef)
  configuration: --prefix=/ffbuild/prefix --pkg-config-flags=--static --pkg-config=pkg-config --cross-prefix=x86_64-w64-mingw32- --arch=x86_64 --target-os=mingw32 --enable-gpl --enable-version3 --disable-debug --disable-w32threads --enable-pthreads --enable-iconv --enable-libxml2 --enable-zlib --enable-libfreetype --enable-libfribidi --enable-gmp --enable-lzma --enable-fontconfig --enable-libvorbis --enable-opencl --disable-libpulse --enable-libvmaf --disable-libxcb --disable-xlib --enable-amf --enable-libaom --enable-avisynth --enable-libdav1d --enable-libdavs2 --disable-libfdk-aac --enable-ffnvcodec --enable-cuda-llvm --enable-frei0r --enable-libgme --enable-libass --enable-libbluray --enable-libmp3lame --enable-libopus --enable-librist --enable-libtheora --enable-libvpx --enable-libwebp --enable-lv2 --enable-libmfx --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenh264 --enable-libopenjpeg --enable-libopenmpt --enable-librav1e --enable-librubberband --enable-schannel --enable-sdl2 --enable-libsoxr --enable-libsrt --enable-libsvtav1 --enable-libtwolame --enable-libuavs3d --disable-libdrm --disable-vaapi --enable-libvidstab --enable-vulkan --enable-libshaderc --enable-libplacebo --enable-libx264 --enable-libx265 --enable-libxavs2 --enable-libxvid --enable-libzimg --enable-libzvbi --extra-cflags=-DLIBTWOLAME_STATIC --extra-cxxflags= --extra-ldflags=-pthread --extra-ldexeflags= --extra-libs=-lgomp --extra-version=20220415
  libavutil      57. 24.101 / 57. 24.101
  libavcodec     59. 26.100 / 59. 26.100
  libavformat    59. 22.100 / 59. 22.100
  libavdevice    59.  6.100 / 59.  6.100
  libavfilter     8. 33.100 /  8. 33.100
  libswscale      6.  6.100 /  6.  6.100
  libswresample   4.  6.100 /  4.  6.100
  libpostproc    56.  5.100 / 56.  5.100
Input #0, matroska,webm, from 'Auswandern - 49.  Landwirte, Landwirtschaft, Investitionen [W4Vvua7H7zs].webm':
  Metadata:
    title           : Auswandern: 49.  Landwirte, Landwirtschaft, Investitionen
    SYNOPSIS        : Auswandern 2022 -2025 bei Youtube: https://www.youtube.com/channel/UCZ4zZkE8REK8Ja805VNfl6w - Paraguay, Brasilien, Uruguay, Argentinien, Georgien ...
                    :
                    : Wie und wo lohnen sich Investitionen in der Landwirtschaft?
    ARTIST          : Auswandern 2022 - 25
    PURL            : https://www.youtube.com/watch?v=W4Vvua7H7zs
    HTTP://YOUTUBE.COM/STREAMING/OTF/DURATIONS/112015: Segment-Count: 80
                    : Segment-Durations-Ms: 5339,5338,5339(r=1),5338,5339(r=1),5338,5339(r=1),5338,5339(r=1),5338,5339(r=1),5338,5339(r=1),5338,5339(r=1),5338,5339(r=1),5338,5339(r=1),5338,5339(r=1),5338,5339(r=1),5338,5339(r=1),5338,5339(r=1),5338,5339(r=1),5338,5339(r=1),533
                    :
                    :
    COMMENT         : https://www.youtube.com/watch?v=W4Vvua7H7zs
    DATE            : 20211012
    DESCRIPTION     : Auswandern 2022 -2025 bei Youtube: https://www.youtube.com/channel/UCZ4zZkE8REK8Ja805VNfl6w - Paraguay, Brasilien, Uruguay, Argentinien, Georgien ...
                    :
                    : Wie und wo lohnen sich Investitionen in der Landwirtschaft?
    ENCODER         : Lavf59.20.101
  Duration: 00:07:05.54, start: -0.007000, bitrate: 1953 kb/s
  Stream #0:0(eng): Video: vp9 (Profile 0), yuv420p(tv, bt709/unknown/unknown), 1920x1080, SAR 1:1 DAR 16:9, 29.97 fps, 29.97 tbr, 1k tbn (default)
    Metadata:
      DURATION        : 00:07:05.539000000
  Stream #0:1(eng): Audio: opus, 48000 Hz, stereo, fltp (default)
    Metadata:
      DURATION        : 00:07:05.541000000
```

**Comment by pboettcher** [comment]
I used your command line to download file both in Windows 7 with yt-dlp 2022.03.08.1 and in CentOS with yt-dlp 2022.04.08.

Then I checked both files with Windows-version of ffprobe. The file downloaded in Windows really contains now all the metadata while the file downloaded in CentOS doesn't.

**Windows:**

ffprobe version git-2020-03-15-c467328 Copyright (c) 2007-2020 the FFmpeg developers
  built with gcc 9.2.1 (GCC) 20200122
  configuration: --enable-gpl --enable-version3 --enable-sdl2 --enable-fontconfig --enable-gnutls --enable-iconv --enable-libass --enable-libdav1d --enable-libbluray --enable-libfreetype --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-libopus --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libtheora --enable-libtwolame --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libzimg --enable-lzma --enable-zlib --enable-gmp --enable-libvidstab --enable-libvorbis --enable-libvo-amrwbenc --enable-libmysofa --enable-libspeex --enable-libxvid --enable-libaom --enable-libmfx --enable-ffnvcodec --enable-cuda-llvm --enable-cuvid --enable-d3d11va --enable-nvenc --enable-nvdec --enable-dxva2 --enable-avisynth --enable-libopenmpt --enable-amf
  libavutil      56. 42.100 / 56. 42.100
  libavcodec     58. 75.100 / 58. 75.100
  libavformat    58. 41.100 / 58. 41.100
  libavdevice    58.  9.103 / 58.  9.103
  libavfilter     7. 77.100 /  7. 77.100
  libswscale      5.  6.101 /  5.  6.101
  libswresample   3.  6.100 /  3.  6.100
  libpostproc    55.  6.100 / 55.  6.100
Input #0, matroska,webm, from 'Auswandern - 49.  Landwirte, Landwirtschaft, Investitionen [W4Vvua7H7zs]_video.webm':
  Metadata:
    title           : Auswandern: 49.  Landwirte, Landwirtschaft, Investitionen
    COMMENT         : https://www.youtube.com/watch?v=W4Vvua7H7zs
    HTTP://YOUTUBE.COM/STREAMING/OTF/DURATIONS/112015: Segment-Count: 80 
                    : Segment-Durations-Ms: 5339,5338,5339(r=1),5338,5339(r=1),5338,5339(r=1),5338,5339(r=1),5338,5339(r=1),5338,5339(r=1),5338,5339(r=1),5338,5339(r=1),5338,5339(r=1),5338,5339(r=1),5338,5339(r=1),5338,5339(r=1),5338,5339(r=1),5338,5339(r=1),5338,5339(r=1),533 
                    :  
                    : 
    ARTIST          : Auswandern 2022 - 25
    DATE            : 20211012
    DESCRIPTION     : Auswandern 2022 -2025 bei Youtube: https://www.youtube.com/channel/UCZ4zZkE8REK8Ja805VNfl6w - Paraguay, Brasilien, Uruguay, Argentinien, Georgien ...
                    : 
                    : Wie und wo lohnen sich Investitionen in der Landwirtschaft?
    SYNOPSIS        : Auswandern 2022 -2025 bei Youtube: https://www.youtube.com/channel/UCZ4zZkE8REK8Ja805VNfl6w - Paraguay, Brasilien, Uruguay, Argentinien, Georgien ...
                    : 
                    : Wie und wo lohnen sich Investitionen in der Landwirtschaft?
    PURL            : https://www.youtube.com/watch?v=W4Vvua7H7zs
    ENCODER         : Lavf58.41.100
  Duration: 00:07:05.54, start: -0.007000, bitrate: 1953 kb/s
    Stream #0:0(eng): Video: vp9 (Profile 0), yuv420p(tv, bt709/unknown/unknown), 1920x1080, SAR 1:1 DAR 16:9, 29.97 fps, 29.97 tbr, 1k tbn, 1k tbc (default)
    Metadata:
      DURATION        : 00:07:05.532000000
    Stream #0:1(eng): Audio: opus, 48000 Hz, stereo, fltp (default)
    Metadata:
      DURATION        : 00:07:05.541000000

**CentOS:**

ffprobe version git-2020-03-15-c467328 Copyright (c) 2007-2020 the FFmpeg developers
  built with gcc 9.2.1 (GCC) 20200122
  configuration: --enable-gpl --enable-version3 --enable-sdl2 --enable-fontconfig --enable-gnutls --enable-iconv --enable-libass --enable-libdav1d --enable-libbluray --enable-libfreetype --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-libopus --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libtheora --enable-libtwolame --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libzimg --enable-lzma --enable-zlib --enable-gmp --enable-libvidstab --enable-libvorbis --enable-libvo-amrwbenc --enable-libmysofa --enable-libspeex --enable-libxvid --enable-libaom --enable-libmfx --enable-ffnvcodec --enable-cuda-llvm --enable-cuvid --enable-d3d11va --enable-nvenc --enable-nvdec --enable-dxva2 --enable-avisynth --enable-libopenmpt --enable-amf
  libavutil      56. 42.100 / 56. 42.100
  libavcodec     58. 75.100 / 58. 75.100
  libavformat    58. 41.100 / 58. 41.100
  libavdevice    58.  9.103 / 58.  9.103
  libavfilter     7. 77.100 /  7. 77.100
  libswscale      5.  6.101 /  5.  6.101
  libswresample   3.  6.100 /  3.  6.100
  libpostproc    55.  6.100 / 55.  6.100
Input #0, matroska,webm, from 'Auswandern - 49.  Landwirte, Landwirtschaft, Investitionen [W4Vvua7H7zs]_video.webm':
  Metadata:
    title           : Auswandern: 49.  Landwirte, Landwirtschaft, Investitionen
    encoder         : Lavf56.40.101
  Duration: 00:07:05.54, start: -0.007000, bitrate: 1953 kb/s
    Stream #0:0(eng): Video: vp9 (Profile 0), yuv420p(tv, bt709/unknown/unknown), 1920x1080, SAR 1:1 DAR 16:9, 29.97 fps, 29.97 tbr, 1k tbn, 1k tbc (default)
    Stream #0:1(eng): Audio: opus, 48000 Hz, stereo, fltp (default)
